### PR TITLE
ignore name mismatch when IgnoreInvalidName is set

### PR DESCRIPTION
**Pull request by wfurt**
contributes to https://github.com/dotnet/runtime/issues/71191, adds test for  issue fixed by #72326

I bump to this when I was trying to improve test coverage and I wanted to override authentication errors using newly added policy. 
The spirit of #71191 was ability to override inside SslStream validation magic to customize validation. 
While the name validation is out of the Chain itself, I hook it up so X509VerificationFlags.IgnoreInvalidName flag is respected. 

Aside from this, this plugs some test gaps. 
GenerateCertificates did come from SslStream and Quic start using it as convenience. 
I did not want to add another project reference to HTTP so I moved it to Common and updated Quic and Http to use it.
I will follow-up for SslStream but I did not want to make this change too big.

### CODE REVIEW

**Review by rzikm — APPROVED** [review]
There are some build errors but otherwise LGTM if fixed and clean CI

File: src/libraries/Common/tests/System/Net/Configuration.Certificates.cs [hunk 1]
```diff
@@ -16,6 +18,30 @@ public static partial class Configuration
     {
         public static partial class Certificates
         {
+            private static readonly X509KeyUsageExtension s_eeKeyUsage =
+            new X509KeyUsageExtension(
+                X509KeyUsageFlags.DigitalSignature | X509KeyUsageFlags.KeyEncipherment | X509KeyUsageFlags.DataEncipherment,
+                critical: false);
```
Comment (stephentoub) [inline src/libraries/Common/tests/System/Net/Configuration.Certificates.cs:24]: Nit: spacing
```suggestion
                new X509KeyUsageExtension(
                    X509KeyUsageFlags.DigitalSignature | X509KeyUsageFlags.KeyEncipherment | X509KeyUsageFlags.DataEncipherment,
                    critical: false);
```

File: src/libraries/Common/tests/System/Net/Configuration.Certificates.cs [hunk 2]
```diff
@@ -110,6 +146,115 @@ public static X509Certificate2 GetSelfSigned13ServerCertificate()
...
+
+            public static void CleanupCertificates([CallerMemberName] string? testName = null, StoreName storeName = StoreName.CertificateAuthority)
+            {
+                string caName = $"O={testName}";
+                try
+                {
+                    using (X509Store store = new X509Store(storeName, StoreLocation.LocalMachine))
+                    {
+                        store.Open(OpenFlags.ReadWrite);
+                        foreach (X509Certificate2 cert in store.Certificates)
```
Comment (stephentoub) [inline src/libraries/Common/tests/System/Net/Configuration.Certificates.cs:158]: All of these need to be disposed of, right? Same for the iteration below.

File: src/libraries/Common/tests/System/Net/Configuration.Certificates.cs [hunk 3]
```diff
@@ -110,6 +146,115 @@ public static X509Certificate2 GetSelfSigned13ServerCertificate()
...
+                extensions.Add(s_eeConstraints);
+                extensions.Add(s_eeKeyUsage);
+                extensions.Add(serverCertificate ? s_tlsServerEku : s_tlsClientEku);
+
+                return extensions;
+            }
+
+            public static (X509Certificate2 certificate, X509Certificate2Collection) GenerateCertificates(string targetName, [CallerMemberName] string? testName = null, bool longChain = false, bool serverCertificate = true)
+            {
+                const int keySize = 2048;
```
Comment (stephentoub) [inline src/libraries/Common/tests/System/Net/Configuration.Certificates.cs:209]: Nit: keySize => KeySize

File: src/libraries/System.Net.Http/tests/FunctionalTests/SocketsHttpHandlerTest.cs [hunk 4]
```diff
@@ -3728,6 +3730,132 @@ public sealed class SocketsHttpHandler_RequestContentLengthMismatchTest_Http3 :
         protected override Version UseVersion => HttpVersion.Version30;
     }
 
+    public abstract class SocketsHttpHandler_SslTest : HttpClientHandlerTestBase
```
Comment (stephentoub) [inline src/libraries/System.Net.Http/tests/FunctionalTests/SocketsHttpHandlerTest.cs:3733]: Nit: SslTest => SslOptionsTest? (If you change it, also change the derived type names)

Comment (wfurt) [reply src/libraries/System.Net.Http/tests/FunctionalTests/SocketsHttpHandlerTest.cs:3733]: I was not sure. The intention is to test security related functions e.g. SslStream, Quic & certificate handling...

**Pull request merged**